server/routes/api/chart.py

```diff
@@ -110,6 +126,70 @@ def get_landing_page_data(dcid):
     )
     return response
 
+def get_date_to_add_for_chart(chart, sv_data):
+    """Get the date for which data is needed for a specific chart
```

Comment:
Could you add more description here? Which dates end up being picked?

Reply:
I'm also surprised we are not inspecting the chart at all - different chart types will have different requirements (some latest, some all dates). Perhaps we can continue ignoring given that we are going to be generating the multiple types of charts now per stat var, but we should rename this function in that case.

Reply:
You're right, the name is a little misleading. This is a function just to find the latest common date for a chart. We are not inspecting the chart because if a stat var needs all dates, there is a separate check for that elsewhere in the code.